x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/service.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  ExceptionListItemSchema,
+  FoundExceptionListItemSchema,
+} from '@kbn/securitysolution-io-ts-list-types';
+import { ENDPOINT_HOST_ISOLATION_EXCEPTIONS_LIST_ID } from '@kbn/securitysolution-list-constants';
+import { HttpStart } from 'kibana/public';
+import { EXCEPTION_LIST_ITEM_URL, EXCEPTION_LIST_URL } from '../event_filters/constants';
+import { HOST_ISOLATION_EXCEPTIONS_LIST } from './constants';
+
+async function createHostIsolationExceptionList(http: HttpStart) {
+  try {
+    await http.post<ExceptionListItemSchema>(EXCEPTION_LIST_URL, {
+      body: JSON.stringify(HOST_ISOLATION_EXCEPTIONS_LIST),
+    });
+  } catch (err) {
+    // Ignore 409 errors. List already created
+    if (err.response?.status !== 409) {
+      throw err;
+    }
+  }
+}
+
+let listExistsPromise: Promise<void>;
```

Comment:
Looking at this local module variable that is global to the module and the above error handling which doesn't do retry logic on errors but just blows up the front end? This makes me wonder if maybe you should consider some of the RxJS utilities?

https://www.learnrxjs.io/learn-rxjs/operators/error_handling/retry

There's examples in the code base which use `retry` and `retryWhen`
https://github.com/elastic/kibana/blob/master/x-pack/plugins/alerting/server/health/get_state.ts#L91

I'm not a big RxJS person so I can't say much more and feel bad suggesting something I myself haven't used, but I think that RxJS should be considered here as an alternative. You might be able to get retry logic on errors, a cache like you want from the result that you did indeed get a success that the list is created from the front end and learn a bit more about the functional side of RxJS and how others in the company use it.

Good excuse to play with something neat? ;-)